spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/SparkZOrderStrategy.java

```diff
@@ -133,6 +133,14 @@ public SparkZOrderStrategy(Table table, SparkSession spark, List<String> zOrderC
           "Cannot perform ZOrdering, all columns provided were identity partition columns and cannot be used.");
     }
 
+    zOrderColNames.forEach(col -> {
+      NestedField nestedField = table.schema().caseInsensitiveFindField(col);
```

Comment:
This should use the case sensitivity setting from Spark, rather than always using case insensitive.